src/format.rs

```diff
@@ -41,36 +41,63 @@ impl fmt::Display for FormattedDuration {
     }
 }
 
+// `HumanDuration` should be as intuitively understandable as possible.
+// So we want to round, not truncate (otherwise 1 hour and 59 minutes
+// would display an ETA of "1 hour" which underestimates the time remaining
+// by a factor 2.
+//
+// To make the precision more uniform, we avoid displaying "1 unit"
+// (except for seconds), because it would be displayed for an important
+// time relatively to the unit itself.  Instead, when we arrive around
+// 1.5 unit, we change from "2 units" to the next smaller unit
+// (e.g. "89 seconds").
+//
+// Formally:
+// * for n >= 2, we go from "n+1 units" to "n units" exactly at (n + 1/2) units
+// * we switch from "2 units" to the next smaller unit at (1.5 unit minus half of the next smaller unit)
+
 impl fmt::Display for HumanDuration {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let t = self.0.as_secs();
         let alt = f.alternate();
-        macro_rules! try_unit {
-            ($secs:expr, $sg:expr, $pl:expr, $s:expr) => {
-                let cnt = t / $secs;
-                if cnt == 1 {
-                    if alt {
-                        return write!(f, "{}{}", cnt, $s);
-                    } else {
-                        return write!(f, "{} {}", cnt, $sg);
-                    }
-                } else if cnt > 1 {
-                    if alt {
-                        return write!(f, "{}{}", cnt, $s);
-                    } else {
-                        return write!(f, "{} {}", cnt, $pl);
-                    }
+        // FIXME when `Duration::SECOND` is stable
+        let second = Duration::from_secs(1);
+        let minute = 60 * second;
+        let hour = 60 * minute;
+        let day = 24 * hour;
+        let week = 7 * day;
+        let year = 365 * day;
+        // FIXME when `div_duration_f64` is stable
+        let t = self.0.as_secs_f64();
+        if self.0 + week / 2 >= year + year / 2 {
```

Comment:
There's a lot of repetition here. I'm pretty sure this could be written as a loop given the right const input tuples...